dashboard/modules/reporter/reporter_head.py

```diff
@@ -112,6 +123,209 @@ async def get_cluster_status(self, req):
                 cluster_status=debug_status(formatted_status_string, error),
             )
 
+    """
+    We don't use pid and ip to get task traceback since ip and pid is bounded to a worker
+    and a worker may run different tasks at different time.
+    Therefore, we may provide wrong traceback info since we don't know which task is running on the worker
+
+    Raises:
+        ValueError: HTTPInternalServerError
+    """
+
+    @routes.get("/task/traceback")
+    async def get_task_traceback(self, req) -> aiohttp.web.Response:
+        logger.info(f"req {type(req)}: {req}")
+        logger.info("in get_traceback")
+        if "task_id" not in req.query:
+            raise ValueError("task_id is required")
+        if "attempt_number" not in req.query:
+            raise ValueError("task's attempt number is required")
+        task_id = req.query.get("task_id")
+        attempt_number = req.query.get("attempt_number")
+        option = ListApiOptions(
+            filters=[
+                ("task_id", "=", task_id),
+                ("attempt_number", "=", attempt_number),
+            ],
+            detail=True,
+            timeout=10,
+        )
+        gcs_channel = self._dashboard_head.aiogrpc_gcs_channel
+        self._state_api_data_source_client = StateDataSourceClient(
+            gcs_channel, self._dashboard_head.gcs_aio_client
+        )
+        self._state_api = StateAPIManager(self._state_api_data_source_client)
+        logger.info(f"self._state_api {type(self._state_api)}: {self._state_api}")
+
+        result = await self._state_api.list_tasks(option=option)
+        tasks = result.result
+        logger.info(f"tasks {type(tasks)}: {tasks}")
+
+        pid = tasks[0]["worker_pid"]
+        node_id = tasks[0]["node_id"]
+        ip = DataSource.node_id_to_ip[node_id]
+        logger.info(f"pid {type(pid)}: {pid}")
+        logger.info(f"node_id {type(node_id)}: {node_id}")
+
+        # Default not using `--native` for profiling
+        native = req.query.get("native", False) == "1"
+        logger.info(
+            "Sending stack trace request to {}:{} with native={}".format(
+                req.query.get("ip"), pid, native
+            )
+        )
+        reporter_stub = self._stubs[ip]
+        reply = await reporter_stub.GetTraceback(
+            reporter_pb2.GetTracebackRequest(pid=pid, native=native)
+        )
```

Comment:
Don't we have to check if the task is still running before we send the traceback request too?

Reply:
Although we would return error from `reporter_stub.GetTraceback` if the task is not running, we should early throw the error before calling the function to save time